openhands/core/main.py

```diff
@@ -202,6 +205,14 @@ async def on_event(event: Event):
     await controller.close()
     state = controller.get_state()
 
+    # save trajectories if applicable
+    if config.trajectories_path is not None:
+        file_path = os.path.join(config.trajectories_path, sid + '.traj')
+        os.makedirs(os.path.dirname(file_path), exist_ok=True)
+        histories = [event_to_dict(event) for event in state.history.get_events()]
+        with open(file_path, 'w') as f:
+            json.dump(histories, f)
```

Comment:
Just wondering, does this address a different need than configuring the `file_store` of the event stream to be any other than the default (the default is in-memory), then it will save every event there?
e.g. personally I use quite a lot something like
```
file_store="local"
file_store_path="/opt/repos/openhands/cache"
```

I can think of a difference though: the full event stream will include some extra stuff like NullObs, which `state.history.get_events()` doesn't return.

Reply:
Yeah I see two differences here:

1) File store is meant to persistent "files", not just "event stream", even though event stream might be the only thing that is stored currently. In theory it can store many other things. Imagine OpenHands supports "snapshots of workspace" in the future... Users may not want to persist the file store just because they need the trajectory.
2) As you pointed out, event stream and trajectories are still different as in event stream contains more things that are useful to the system but less interesting to end users.

Reply:
You do have a point. Strictly speaking, the file store also stores the agent state, in a separate pickled file.